### DISCUSSION TITLE
How to read nested zips?

### DISCUSSION THREAD

**Discussion by nk9** [Toblerity/Fiona#1318] [Q&A]
I have a nested zip that looks like this:

```
level1.zip/level1/level2.zip/level2/level3.zip/level3/levelThree.shp
```

I couldn't really figure out how to use `ZipMemoryFile` with this setup; the nesting is doing my head in. I actually have code to pull out all the `.{shp,dbf,prj}` files from the level3 zip using nested calls to Python's `zipfile` built-in module, so if I could find a way to just hand all three to Fiona as bytes, then I'd be off to the races. I'm transitioning from pyshp; there, you can just do `shapefile.Reader(dbf=zip.open('path.dbf'), prj=zip.open('path.prj'), shp=zip.open('path.shp'))`. If something like this is possible, that would be great.

In searching for a solution, I found GDAL's [`/vsizip/` syntax](https://gdal.org/user/virtual_file_systems.html#vsizip-zip-archives). It actually SEEMS not crazily complex, and purports to do exactly what I want. [This post](https://lists.osgeo.org/pipermail/gdal-dev/2016-October/045259.html) gave me hope… So I tried this:

```
import fiona

colxn = fiona.open(
    "/vsizip/vsizip/vsizip//Users/nick/level1.zip/level1/level2.zip/level2/level3.zip/level3/levelThree.shp"
)
print(colxn)
```

But no dice:

> fiona.errors.DriverError: '/vsizip/vsizip/vsizip//Users/nick/level1.zip/level1/level2.zip/level2/level3.zip/level3/levelThree.shp' does not exist in the file system, and is not recognized as a supported dataset name.

Should this work, or am I doing something wrong?

It's a little less robust than the first approach, because I'm assuming that the file structure inside doesn't change. I'd really prefer the first way, recursively listing things and then passing all the necessary files opened in memory to fiona.

Thanks in advance!

**Comment by nk9** [comment]
Would you belive it: I actually had my path wrong on disk, even before the first zip file. 😊  When I fixed that, the code above works! I'm going to leave it here because I can find precious little on using this `vsizip` syntax. 

I'd still be interested to hear if it's possible to pass Fiona several different bytes-like objects representing the various sidecars and have it read them all.

**Comment by sgillies** [comment]
@nk9 I've never needed to work with nested zip files. I'm not surprised that multiple /vsizip/ works despite lack of documentation. Distributing data as nested zip files seems like a bad idea to me. I wouldn't want Fiona to explicitly support them and thus propagate the practice.
